restli-server/src/test/java/com/linkedin/restli/internal/server/model/TestRestLiApiBuilder.java

```diff
@@ -198,4 +198,54 @@ public void testActionReturnType(Class<?> resourceClass, Class<?> expectedAction
       Assert.assertEquals(resourceMethodDescriptor.getActionReturnType(), expectedActionReturnType);
     }
   }
+
+  @DataProvider(name = "unsupportedFinderResourceTypeData")
+  private Object[][] unsupportedFinderResourceTypeData()
+  {
+    return new Object[][]
+        {
+            { FinderUnsupportedKeyUnstructuredDataResource.class, "KeyUnstructuredDataResource class does not support @Finder methods" },
+            { FinderUnsupportedSingleUnstructuredDataResource.class, "SingleUnstructuredDataResource class does not support @Finder methods" }
+        };
+  }
+
+  /**
+   * Ensures that when finder methods are processed, if the return type is not of a Record, then it will be warned.
```

Comment:
I re-write it to "Ensures that when finder methods are processed, when the resource value class is a SingleUnstructuredDataResource or a KeyUnstructuredDataResource, we will end up with a ResourceConfigException because we don't support that righ tnow." Hope that seems like a good doc string.